wandb/sdk/launch/runner/kubernetes_runner.py

```diff
@@ -600,6 +645,51 @@ def inject_entrypoint_and_args(
             containers[i]["command"] = entry_point.command
 
 
+async def ensure_api_key_secret(
+    core_api: "CoreV1Api",
+    secret_name: str,
+    namespace: str,
+    api_key: str,
+) -> "V1Secret":
+    """Create a secret containing a user's wandb API key.
+
+    Arguments:
+        core_api: The Kubernetes CoreV1Api object.
+        namespace: The namespace to create the secret in.
+        api_key: The user's wandb API key
+
+    Returns:
+        The created secret
+    """
+    secret_data = {"password": base64.b64encode(api_key.encode()).decode()}
+    secret = client.V1Secret(
+        data=secret_data,
+        metadata=client.V1ObjectMeta(name=secret_name, namespace=namespace),
+        kind="Secret",
+        type="kubernetes.io/basic-auth",
+    )
+
+    try:
+        try:
+            return await core_api.create_namespaced_secret(namespace, secret)
+        except ApiException as e:
+            # 409 = conflict = secret already exists
+            if e.status == 409:
+                existing_secret = await core_api.read_namespaced_secret(
+                    name=secret_name, namespace=namespace
+                )
+                if existing_secret.data != secret_data:
+                    raise LaunchError(
```

Comment:
I commented in the helm charts PR re: adding a label to distinguish it from secrets created by this function.  with this labeling in place, I think in this case we can attempt to *replace* the stale secret instead of erroring out IFF the existing secret bears the label that indicates it was created by the agent.

we would still want to error out if the existing secret is managed by helm, because the agent should not be mucking around with helm-managed resources (doing so would cause cluster state to drift from whatever helm commanded).

another reason to do this is to prevent the cluster from latching into a bad state.  it's not obvious at all that the k8s runner passes the api key via secret, and the user might get confused if a previous agent exited before it was able to clean up the secret and they attempt to restart the agent with a new api key.